src/mdns.c

```diff
@@ -354,15 +354,41 @@ mdns_resolve(struct mdns_ctx *ctx, const char *addr, unsigned short port)
         struct sockaddr_storage *mcast_addrs = NULL;
         size_t i;
         int status;
+        int family;
 
         sprintf(buf, "%hu", port);
         memset(&hints, 0, sizeof(hints));
-        hints.ai_family = AF_UNSPEC;
         hints.ai_socktype = SOCK_DGRAM;
         hints.ai_flags = AI_NUMERICHOST | AI_NUMERICSERV;
-        errno = getaddrinfo(addr, buf, &hints, &res);
-        if (errno != 0)
+
+        if (addr == NULL) {
+            family = AF_UNSPEC;
+            hints.ai_family = family;
+            /* First, get the ipv4 multicast address info */
+            errno = getaddrinfo(MDNS_ADDR_IPV4, buf, &hints, &res);
+            if (errno != 0)
+                    return (MDNS_LKPERR);
```

Comment:
Can one just have one IPv6 address and not an IPv4 one?

Reply:
That's a valid point. I'll fix this once I'm done with windows fixes

Reply:
I was expecting some issues with a `getaddrinfo()` failure, I tried disabling ipv6 entirely on my system, `getaddrinfo()` appears to be returning some valid records anyway. 

However, at a later stage when the interfaces are probed, they won't be ipv4/6 enabled and won't be taken into account (https://github.com/videolabs/libmicrodns/blob/6e96f538d3f99e452194e8711b05f88d9b966d72/src/mdns.c#L97)
If dual stack is enabled, it will use the other protocol, if the user explicitely requested a protocol version but they don't have any adapter using that protocol, it will gracefully fail to initialize

TL;DR I think the current code is ok